vendor/github.com/ethereum/go-ethereum/whisper/whisperv6/whisper.go

```diff
@@ -821,8 +825,24 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 					log.Warn("failed to decode p2p request message, peer will be disconnected", "peer", p.peer.ID(), "err", err)
 					return errors.New("invalid p2p request")
 				}
+
 				whisper.mailServer.DeliverMail(p, &request)
 			}
+		case p2pRequestAckCode:
+			if p.trusted {
+				var envelope Envelope
+				if err := packet.Decode(&envelope); err != nil {
```

Comment:
it doesnt have to be envelope, any payload will work. common.Hash for example

Reply:
@dshulyak good point thank you! I left it after changing this part but now it can be removed